src/components/DataTable/DataTable.tsx

```diff
@@ -52,19 +54,26 @@ const extractIconFromCell = (cellValue: unknown): React.ReactElement | undefined
 export const MoonstoneTable = <T extends Record<string, unknown>>({
     data,
     columns,
-    isStructured = false
+    isStructured = false,
+    enableRowSelection = false
 }: DataTableProps<T>) => {
     const [expanded, setExpanded] = useState<ExpandedState>({});
+    const [rowSelection, setRowSelection] = useState<RowSelectionState>({});
 
     const tableColumns = createTableColumns(columns);
 
     const table = useReactTable({
         data,
         columns: tableColumns,
+        state: {
+            expanded,
+            rowSelection
+        },
+        onExpandedChange: setExpanded,
```

Comment:
Why do you move `onExpandedChange` outside the `isStructured` condition?
It was a good idea to use it only when the data is structured.